src/nix/deployment/mod.rs

```diff
@@ -671,6 +671,26 @@ impl Deployment {
             target
         };
 
+        // Delete old generations
+        let mut target = if self.options.delete_old {
+            let job = parent.create_job(JobType::DeleteOld, nodes.clone())?;
+            job.run_waiting(|job| async move {
+                job.state(JobState::Running)?;
+                job.message("Deleting old NixOS generations...".to_string())?;
+
+                let host = target.host.as_mut().unwrap();
+                host.set_job(Some(job.clone()));
+                host.run_command(&["nix-collect-garbage", "--delete-old"])
```

Comment:
This deletes all old generations.

Using '--delete-older-than' and requiring the period to be provided via the colmena CLI argument allows for a more fine-grained deletion approach, including deleting all.